Cache option for Windows

**Description**

Using `git config --global gitsign.connectorID https://github.com/login/oauth` with gitsign and then perform multiple commits can result in reaching the [secondary rate limit](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#about-secondary-rate-limits) on calling the GitHub repository. Current cache option doesn't work on Windows as mention in #194.

![image](https://github.com/sigstore/gitsign/assets/3404761/3f14a556-60cc-4d78-97c9-a35335e02967)

**Version**

```
gitsign version v0.10.1
parsed config:
{
  "Fulcio": "https://fulcio.sigstore.dev",
  "FulcioRoot": "",
  "Rekor": "https://rekor.sigstore.dev",
  "RekorMode": "online",
  "ClientID": "sigstore",
  "RedirectURL": "",
  "Issuer": "https://oauth2.sigstore.dev/auth",
  "ConnectorID": "https://github.com/login/oauth",
  "TokenProvider": "",
  "TimestampURL": "",
  "TimestampCert": "",
  "LogPath": "",
  "CommitterName": "Foo",
  "CommitterEmail": "Bar",
  "MatchCommitter": false,
  "Autoclose": true,
  "AutocloseTimeout": 6
}
```

I'm open to this, but I do not have any sort of Windows socket experience (or much experience building Windows apps in general). 😓 

The first step would be to hide unix-specific syscalls behind a build flag:

```sh
$ GOOS=windows go build ./cmd/gitsign-credential-cache
# github.com/sigstore/gitsign/cmd/gitsign-credential-cache
./main.go:43:10: undefined: syscall.Umask
```

--- a/.goreleaser.yaml
+++ b/.goreleaser.yaml
@@ -36,7 +36,7 @@ builds:
       - linux
       - darwin
       - freebsd
-      # - windows # TODO: fix undefined: syscall.Umask for windows builds
+      - windows
     goarch:
       - amd64
       - arm64

--- a/cmd/gitsign-credential-cache/main.go
+++ b/cmd/gitsign-credential-cache/main.go
@@ -21,7 +21,6 @@ import (
 	"net/rpc"
 	"os"
 	"path/filepath"
-	"syscall"
 
 	"github.com/coreos/go-systemd/v22/activation"
 	"github.com/spf13/pflag"
@@ -38,9 +37,6 @@ var (
 
 func main() {
 	pflag.Parse()
-	// Override default umask so created files are always scoped to the
-	// current user.
-	syscall.Umask(0077)
 
 	if *versionFlag {
 		v := version.GetVersionInfo()
@@ -93,6 +89,16 @@ func main() {
 		if err != nil {
 			log.Fatalf("error opening socket: %v", err)
 		}
+
+		// Previously, we used syscall.Umask(0077) to ensure this was
+		// permissioned only to the current user. Windows doesn't have this
+		// syscall, so we're switching over to an explicit Chmod on the socket
+		// path.
+		// Also see https://github.com/golang/go/issues/11822
+		if err := os.Chmod(path, 0700); err != nil {
+			log.Fatalf("error setting socket permissions: %v", err)
+		}
+
 		go connToChan(l, connChan)
 	}
 	srv := rpc.NewServer()

